dns_sd_avahi.c

```diff
@@ -14,42 +15,191 @@
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
  * Lesser General Public License for more details.
  *
+ * Some of this is insipred from libavahi's example:
+ * https://avahi.org/doxygen/html/client-browse-services_8c-example.html
+ * which is also LGPL 2.1 or later.
+ *
  * */
 
-#include "iio.h"
+#include "iio-private.h"
 
 #include <errno.h>
 #include <stdbool.h>
 #include <string.h>
 #include <time.h>
+#include <sys/time.h>
+
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netdb.h>
 
 #include <avahi-common/error.h>
 #include <avahi-common/simple-watch.h>
+#include <avahi-common/malloc.h>
 #include <avahi-client/client.h>
 #include <avahi-client/lookup.h>
 
 #include "debug.h"
 
+/* in network.c */
+int create_socket(const struct addrinfo *addrinfo, unsigned int timeout);
+#define DEFAULT_TIMEOUT_MS 5000
+#define IIOD_PORT 30431
```

Comment:
maybe it's worth adding a `network.h` and moving things there; especially if exporting `create_socket()`

Reply:
yeah, wasn't sure - makes the most sense - I assume `network-private.h`?